inc/hal/mcu.hpp

```diff
@@ -0,0 +1,58 @@
+/**
+ * @file mcu.hpp
+ *
+ * @brief MCU related
+ *
+ * @date 03/2024
+ */
+
+#ifndef __MCU_HPP__
+#define __MCU_HPP__
+
+#include <cstdint>
+
+/*****************************************
+ * Public Function Prototypes
+ *****************************************/
+
+extern "C"
+{
+/**
+ * @brief Initializes System Clock
+ * @note  Defined by cube
+ */
+void SystemClock_Config(void);
+}
+
+namespace hal {
+class mcu {
+    public:
+        /**
+         * @brief Initializes MCU and some peripherals
+         */
+        static void init(void);
+
+        /**
+         * @brief Put the MCU to sleep
+         *
+         * @param ms Sleep time in milliseconds
+         */
+        static void sleep(uint32_t ms);
```

Comment:
```suggestion
        static void sleep_ms(uint32_t ms);
```

não acho que vocês vão chegar a usar essa função, mas eh legal ser mais descritivo se pa

Reply:
A gente ainda vai refatorar essa parte do mcu, não sou fão de como ela ta não